framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -471,17 +505,17 @@ def localFinalizeActualSampling(self,jobObject,model,myInput):
             if self.convergeTraj[traj]:
               self.status[traj] = {'process':None, 'reason':'converged'}
             else:
-              # if rejecting bad point, keep the old point as the new point
-              if self.status[traj]['reason'] != 'rejecting bad opt point':
-                try:
-                  self.counter['recentOptHist'][traj][1] = copy.deepcopy(self.counter['recentOptHist'][traj][0])
-                except KeyError:
-                  # this means we don't have an entry for this trajectory yet, so don't copy anything
-                  pass
-                self.counter['recentOptHist'][traj][0] = {'inputs':self.optVarsHist[traj][self.counter['varsUpdate'][traj]],
-                                                          'output':currentObjectiveValue}
               # update status to submitting grad eval points
               self.status[traj]['process'] = 'submitting grad eval points'
+            # if rejecting bad point, keep the old point as the new point; otherwise, add the new one
+            if self.status[traj]['reason'] != 'rejecting bad opt point':
```

Comment:
Okay, that sounds fine.  (I think for string constants, Python actually puts all the identical strings in the module in the same location, and then before comparing the bytes looks and sees if they are the same object first, so I don't think it is much slower than integers.)